src/angular-youtube-embed.js

```diff
@@ -184,13 +184,44 @@ angular.module('youtube-embed', ['ng'])
                 player.id = playerId;
                 return player;
             }
+            
+            function isEmpty (obj) {
+                if (obj == null) return true;
+                if (Object.keys) return Object.keys(obj).length == 0;
+                var keys = [];
+                for (var key in obj) if (_.has(obj, key)) keys.push(key);
+                return keys.length == 0;
+            };
 
             function loadPlayer () {
-                if (scope.videoId || scope.playerVars.list) {
+                // If present, use an existing player
+                if(scope.player && !isEmpty(scope.player) && scope.player.getIframe() && scope.player.cueVideoById === 'function') {
+                    var playerVars = ( playerVars ? playerVars : angular.copy(scope.playerVars) );
```

Comment:
how would `playerVars` be defined here? maybe i'm missing something, but i don't understand this ternary operator